public/background.js

```diff
@@ -271,6 +277,104 @@ const processGetOrdinalsRequest = (sendResponse) => {
   }
 };
 
+const processGetExchangeRate = (sendResponse) => {
+  chrome.storage.local.get(['exchangeRateCache'], async (data) => {
+    try {
+      const { exchangeRateCache } = data;
+      if (exchangeRateCache?.rate && Date.now() - exchangeRateCache.timestamp < 5 * 60 * 1000) {
+        sendResponse({
+          type: 'getExchangeRate',
+          success: true,
+          data: Number(exchangeRateCache.rate.toFixed(2)),
+        });
+        sendResponse();
+      } else {
+        const res = await fetch(`${WOC_BASE_URL}/main/exchangerate`);
+        if (!res.ok) {
+          throw new Error(`Fetch error: ${res.status} - ${res.statusText}`);
+        }
+        const data = await res.json();
+        const rate = data.rate;
+        const currentTime = Date.now();
+        chrome.storage.local.set({ exchangeRateCache: { rate, timestamp: currentTime } });
+        sendResponse({
+          type: 'getExchangeRate',
+          success: true,
+          data: Number(rate.toFixed(2)),
+        });
+      }
+    } catch (error) {
+      sendResponse({
+        type: 'getExchangeRate',
+        success: false,
+        error: JSON.stringify(error),
+      });
+    }
+  });
+};
+
+const processGetPaymentUtxos = async (message, sendResponse) => {
+  if (!message?.params) {
+    sendResponse({
+      type: 'getPaymentUtxos',
+      success: false,
+      error: 'Unknown network or address!',
+    });
+    return;
+  }
+  try {
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ addresses: message.params.addresses }),
+    };
+    const res = await fetch(
+      `${WOC_BASE_URL}/${message.params.network === 'testnet' ? 'test' : 'main'}/addresses/unspent`,
+      requestOptions,
+    );
+    if (!res.ok) {
+      throw new Error(`Fetch error: ${res.status} - ${res.statusText}`);
+    }
+
+    const items = await res.json();
+
+    let utxoData = [];
+    for (const i of items) {
+      const scriptRes = await fetch(
```

Comment:
I was misunderstanding how this was going to work, and that an address was going to be passed in. This makes total sense, but In that case, we really don't need script, and we shouldn't look it up. I was expecting more magic, but this lookup is not necessary as the calling party can generate the script themselves.

Reply:
Makes sense. So it really should just be getPaymentUtxos() with no params. Will cache as well and update when the user launches the app.